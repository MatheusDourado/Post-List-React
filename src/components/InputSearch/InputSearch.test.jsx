import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { InputSearch } from ".";

describe("<InputSearch/>", () => {
	test("should call handleChange function on each key pressed", () => {
		const fn = jest.fn();
		render(<InputSearch handleChange={fn} searchValue="testing..." />);

		const input = screen.getByPlaceholderText(/digite aqui/i);
		const value = "testing";

		userEvent.type(input, value);

		expect(input.value).toBe("testing...");
		expect(fn).toHaveBeenCalledTimes(value.length);
	});

	test("should have a value of searchValue", () => {
		const fn = jest.fn();
		render(<InputSearch handleChange={fn} searchValue={"testing"} />);

		const value = "testing";
		userEvent.type(screen.getByPlaceholderText("Digite aqui..."), value);

		expect(screen.getByPlaceholderText("Digite aqui...").value).toBe(value);
		expect(fn).toHaveBeenCalledTimes(value.length);
	});

	test("should match snaphot", () => {
		const fn = jest.fn();
		const { container } = render(
			<InputSearch handleChange={fn} searchValue="" />
		);

		expect(container.firstChild).toMatchSnapshot();
	});
});
