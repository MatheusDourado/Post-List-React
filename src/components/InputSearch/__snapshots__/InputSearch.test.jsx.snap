// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSearch/> should match snaphot 1`] = `
<div
  class="header-list"
>
  <h2
    class="title-2"
  >
     Post list 
  </h2>
  <input
    class="search"
    placeholder="Digite aqui..."
    type="search"
    value=""
  />
</div>
`;
